lib-impl/src/main/java/org/ergoplatform/appkit/impl/ExplorerAndPoolUnspentBoxesLoader.java

```diff
@@ -0,0 +1,44 @@
+package org.ergoplatform.appkit.impl;
+
+import org.ergoplatform.appkit.BoxOperations;
+import org.ergoplatform.appkit.InputBox;
+import org.ergoplatform.restapi.client.ErgoTransaction;
+import org.ergoplatform.restapi.client.ErgoTransactionInput;
+import org.ergoplatform.restapi.client.Transactions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@link BoxOperations.IUnspentBoxesLoader} implementation to be used with
+ * {@link BoxOperations#withInputBoxesLoader(BoxOperations.IUnspentBoxesLoader)}
+ * <p>
+ * Instead of the default implementation, this one fetches the mempool from the node connected to
+ * and blacklists the inputs so they aren't used for transactions made by {@link BoxOperations}
+ * methods.
+ * <p>
+ * Note that the mempool is fetched on object instantiation, so don't pass references around as
+ * the data will be outdated after some time.
```

Comment:
Oh, this doesn't sound right. It is better to avoid such effectful operations in a constructor.

Reply:
I think we are dealing with Provider/Loader pattern here. It is better to keep the Loader itself stateless, so it only encapsulate the loading logic itself, but doesn't preserve the loaded data (which is another responsibility). The you can share the loader as necessary.

Reply:
That would mean to fetch the data from node on any page, or do an `if (page == 0) { fetch stuff }` workaround. I can change it like that but really think this is hacky.
I had planned to implement transaction chaining as well, and that situation is not even to handle that way. Unconfirmed boxes need to be fetched on the last page, so when explorer returns 0 boxes. To not run into a never ending recursion, the loader must save that it already returned the unconfirmed boxes to not return it again. If you insist on a stateless loader in appkit, I can't implement the feature here but have to do it on wallet-app repo. Okay for me, but won't help others to use the feature.

Reply:
I have now changed the interface, added `prepare()` and `prepareForAddress` so any preparation needed can be done right in time. This is more flexible, but the implementation for pool support still needs to hold a state between the calls to `prepare()` and `loadBoxesPage()`.